pkg/model/bootstrapscript.go

```diff
@@ -167,10 +167,10 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
 			)
 		}
 
-		// credentials needed always when using swift but when using None dns only in control plane
-		passEnvs := true
-		if !strings.HasPrefix(cluster.Spec.ConfigBase, "swift://") && cluster.UsesNoneDNS() && !b.ig.IsControlPlane() {
-			passEnvs = false
+		// credentials needed always in control-plane and when using gossip also in nodes
```

Comment:
I cannot test the behaviour with openstack designate, hopefully it will work. It should not do any dns changes from worker nodes? If kops uses route53, the dns changes are done by dns-controller and that is running always in control-plane?